.github/workflows/test.yaml

```diff
@@ -36,22 +46,43 @@ jobs:
             PACKAGES: libluajit-5.1-dev libluajit-5.1-2 luajit
             FLAVORFLAGS: -DLUAJIT_FRIENDLY_MODE=ON -DENABLE_LUAJIT=ON
       fail-fast: false
-    runs-on: ubuntu-24.04
+    runs-on: ${{ matrix.os }}
     steps:
       - uses: actions/checkout@v3
 
-      - name: Setup common packages
+      - name: Setup Linux packages
+        if: runner.os == 'Linux'
         run: sudo apt install -y clang-15 libclang-common-15-dev ${{ matrix.PACKAGES }}
 
-      - name: Running CMake
+      - name: Setup macOS packages
+        if: runner.os == 'macOS'
+        run: brew install llvm luajit cmake ninja
+
+      - name: Running CMake (Linux)
+        if: runner.os == 'Linux'
         run: >
           cmake -S . -B build -G Ninja -DENABLE_TESTING=ON
           -DCMAKE_C_COMPILER=clang-15 -DCMAKE_CXX_COMPILER=clang++-15
           ${{ matrix.CMAKEFLAGS }} ${{ matrix.FLAVORFLAGS }}
 
-      - name: Building
+      - name: Running CMake (macOS)
+        if: runner.os == 'macOS'
+        run: >
+          cmake -S . -B build -G Ninja -DENABLE_TESTING=ON
+          -DCMAKE_C_COMPILER=/opt/homebrew/opt/llvm/bin/clang
+          -DCMAKE_CXX_COMPILER=/opt/homebrew/opt/llvm/bin/clang++
+          -DLUA_INCLUDE_DIR=/opt/homebrew/include/luajit-2.1
+          -DLUA_LIBRARIES=/opt/homebrew/lib/libluajit-5.1.dylib
+          ${{ matrix.CMAKEFLAGS }} ${{ matrix.FLAVORFLAGS }}
+
+      - name: Building (Linux)
+        if: runner.os == 'Linux'
         run: cmake --build build --parallel $(nproc)
 
+      - name: Building (macOS)
+        if: runner.os == 'macOS'
+        run: cmake --build build --parallel $(sysctl -n hw.ncpu)
```

Comment:
By default `--parallel` use the default number of parallel jobs, which often corresponds to the number of available CPU cores:

> If <jobs> is omitted, CMake will typically use the native build tool's default number of parallel jobs, which often corresponds to the number of available CPU cores. 

See https://cmake.org/cmake/help/latest/manual/cmake.1.html#cmdoption-cmake-build-j

I suppose we can omit `$(nproc)` and use this step on macOS as well.